src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,274 @@
+     * @return Formatted string representing the date of LocalDate in d MMMM yyyy format.
+     * @throws DukeException
+     */
+    public static String parseLocalDate(LocalDate localDate) throws DukeException {
+        try {
+            return localDate.format(DateTimeFormatter.ofPattern("d MMMM yyyy"));
+        } catch (DateTimeException dateTimeException) {
+            throw new DukeException(Ui.exceptionInvalidLocalDate());
+        }
+    }
+
+    /**
+     * Parses a formatted string into LocalDate.
+     * The formatted string must be of the format "dd/mm/yyyy".
+     *
+     * @param dateString Formatted string to be parsed.
+     * @return LocalDate equivalent of the date represented by dateString.
+     * @throws DukeException If dateString is of an invalid date format.
+     * @throws DukeException If dateString represents a date later than "+999999999-12-31" or
+     *                       earlier than "-999999999-01-01".
+     */
+    public static LocalDate toLocalDate(String dateString) throws DukeException {
+        String[] split = dateString.split("[/\\s]");
+
+        // Check if split has 3 elements. If not, this is already an invalid date format.
+        if (split.length != 3) {
+            throw new DukeException(Ui.exceptionInvalidDateTimeFormat());
+        }
+
+        String date = padZeros(split[0], 2);
+        String month = padZeros(split[1], 2);
+        String year = padZeros(split[2], 4);
+
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append(year).append("-");
+        stringBuilder.append(month).append("-");
+        stringBuilder.append(date);
+
+        LocalDate localDate;
+        try {
+            localDate = LocalDate.parse(stringBuilder.toString());
+        } catch (DateTimeParseException dateTimeParseException) {
+            throw new DukeException(Ui.exceptionInvalidDateTimeFormat());
+        }
+
+        return localDate;
+    }
+
+    /**
+     * Parses index provided in user input into an int.
+     * For instance, when user inputs "done 1", calling parseUserNumInput on this input will return 1.
+     * In this case, the input's identifying command is "done".
+     *
+     * @param userInput User's input into Duke chatbot.
+     * @param command The identifying command in user input.
+     * @return An int representing an index.
+     * @throws DukeException If user input for index cannot be parsed into Integer.
+     */
+    public static int parseUserNumInput(String userInput, Commands command) throws DukeException {
+        // Parses integer in user input. Invalid user input could throw NumberFormatException.
+        try {
+            // Add 1 as user's number input is separated from command by 1 space.
+            return Integer.parseInt(userInput.substring(command.getLength() + 1));
+        } catch (NumberFormatException nfe) {
+            // Invalid user input cannot be parsed into Integer.
```

Comment:
Noticed the same issue about comments here! Maybe you could check your other methods to see if the comments are necessary?